### ISSUE TITLE
Version tags becomes dirty when using setuptools_scm

### ISSUE THREAD

**Issue by lars-petter-hauge** [equinor/komodo#68]
Given a repository tag defined  in the `repository.yml`, the correct tag is downloaded as expected.

```
git clone -b 0.0.1 -q --recursive
Note: checking out 'e06b..'.

You are in 'detached HEAD' state. You can look around, make experimental...
```
and later installed,

However, the written `version.py` has a dirty tag, as if something in the repo had been changed:
`0.0.2.dev0+ge06bf49.d20200123`

**Comment by sondreso** [comment]
This is most likely related to https://github.com/equinor/komodo/commit/2b52826db389a8974505400821112f482a7d4ec9, but we should __not__ revert this as that means we install unwanted packages/version.

**Comment by oysteoh** [comment]
And further - most likely due to us reinstalling the packages after changes in code via fixup shebangs.

**Comment by markusdregi** [comment]
I thought that the `shebang`-"fixup" was done after we reinstall?

https://github.com/equinor/komodo/blob/a3dc53e3d1fd77f8ad71f37a89eff588c0f9f4dd/bin/kmd#L114

**Comment by oysteoh** [comment]
Yes... i'm wrong... with setup-py.sh the packages is not reinstalled at all either... so something is changing some files after download and before `pip install . ..` which makes the versions dirty...

**Comment by lars-petter-hauge** [comment]
TLDR: Could this be `scm_tool` writing to `version.py` making the commit dirty?


Just from a glimpse of which packages that are affect (everest, spinningjenny, pyscal..) and which are not (libres, libecl..). I remember something similar with segyviewer. In that case we installed twice via the setup.py file (once for segyviewlib and once for segyviewer). This had to occur in a specific order though, see comment:
```
# Version number, based of the git tag, is written to version.py in setup(name=segyviewlib..).
# The update of version.py makes the commit dirty, which affects use_scm_version.
# setup(..segyviewer..) must be placed before setup(..segyviewlib..), if not, the version nr for
# segyviewer will be bumped and marked as dev.
```

**Comment by oysteoh** [comment]
Check if builds with the new komodo PR merged still having dirty versions. If it has this needs to be further investigated.

**Comment by oysteoh** [comment]
still dirty - needs more investigation

**Comment by dotfloat** [comment]
__Problem__:
pip makes several temporary directories in `/tmp`, one of which is the "build" directory. This directory is a copy of the source root. From here `setup.py` is called. The problem is that RHEL 6 uses `git` version 1.7.1, which seems to have a bug where the `.git/index` file is not updated to reflect the change in directory, and so `git describe --tags --dirty --long` thinks that the directory is dirty. This problem disappears if you use a newer `git` to do anything and then use `git-describe` from 1.7.1.

__Workaround__:
I added a parameter to Jenkins called `PATH+SDPSOFT_GIT` that points to `/prog/sdpsoft/git-2.8.0/bin`. According to https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#withenv-set-environment-variables , the name `PATH+[...]` means prepend value to `PATH`.

__Solution__:
Ideally we should source sdpsoft git in Jenkins' bashrc. _Every_ job needs a newer git, so it makes no sense to repeat this process for anything we might want to run on Jenkins.

**Comment by markusdregi** [comment]
We are not the only consumer of the jenkins user, hence it is rather scary to change this globally. It is a different story when we get started with the `f_komodo` user...

How about doing the workaround for now and base it on `GIT_EXEC`? The environment can be changed for the entire pipeline in: https://github.com/equinor/komodo/blob/456f4192e2ff2769019105972bbf27cff0e90980/vars/komodo.groovy#L137

**Comment by dotfloat** [comment]
We're now using the `f_komodo` user that sources from SDPsoft git. This seems to have solve the issue.
